lib/mongodb.js

```diff
@@ -802,6 +916,41 @@ MongoDB.prototype.count = function count(model, where, options, callback) {
 };
 
 /**
+ * Replace properties for the model instance data
+ * @param {String} model The model name
+ * @param {Object} data The model data
+ * @param {Function} [callback] The callback function
+ */
+MongoDB.prototype.replaceAttributes = function replaceAttrs(model, id, data, options, cb) {
+  var self = this;
+
+  // Check for other operators, sanitize the data obj and make unset the missing data
+  data = self.parseReplaceData(model, data, options);
+
+  if (self.debug) {
+    debug('updateAttributes', model, id, data);
+  }
+  var oid = self.coerceId(model, id);
+  var idName = this.idName(model);
+
+  this.execute(model, 'findAndModify', {_id: oid}, [
+    ['_id', 'asc']
+  ], data, {}, function (err, result) {
+    if (self.debug) {
+      debug('updateAttributes.callback', model, id, err, result);
+    }
+    var object = result && result.value;
+    if (!err && !object) {
+      // No result
+      err = 'No ' + model + ' found for id ' + id;
+    }
+    self.setIdValue(model, object, id);
+    object && idName !== '_id' && delete object._id;
+    cb && cb(err, object);
+  });
+};
```

Comment:
The implementation of `replaceAttributes` and `replaceOrCreate` look mostly identical to me. Is there a way how to extract shared code to a single shared method?